server/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/UserIdConversionEndpointsTests.java

```diff
@@ -96,46 +120,78 @@ public void testBadFilterWithGroup() {
         endpoints.findUsers("groups.display eq \"foo\"", "ascending", 0, 100, false);
     }
 
+    @Test
+    public void testGoodFilter_IncludeInactive() {
+        final String idzId = randomUUID().toString();
+        arrangeCurrentIdentityZone(idzId);
+
+        final String filter = "(username eq \"foo\" and id eq \"bar\") or username eq \"bar\"";
+
+        final List<ScimUser> allScimUsers = new ArrayList<>();
+        for (int i = 0; i < 5; ++i) {
+            final ScimUser scimUser = new ScimUser(randomUUID().toString(), "bar", "Some", "Name");
+            scimUser.setOrigin("idp2");
+            allScimUsers.add(scimUser);
+        }
+        final ScimUser scimUser6 = new ScimUser("bar", "foo", "Some", "Name");
+        scimUser6.setOrigin("idp1");
+        allScimUsers.add(scimUser6);
+        assertThat(allScimUsers).hasSize(6);
+        arrangeScimUsersForFilter(filter, allScimUsers, true, idzId);
+
+        // check different page sizes -> should return all users, since 'includeInactive' is true
+        assertEndpointReturnsCorrectResult(filter, 1, allScimUsers, true);
+        assertEndpointReturnsCorrectResult(filter, 2, allScimUsers, true);
+        assertEndpointReturnsCorrectResult(filter, 3, allScimUsers, true);
+        assertEndpointReturnsCorrectResult(filter, 4, allScimUsers, true);
+        assertEndpointReturnsCorrectResult(filter, 10, allScimUsers, true);
+    }
+
+    @Test
+    public void testGoodFilter_OnlyActive() {
+        final String idzId = randomUUID().toString();
+        arrangeCurrentIdentityZone(idzId);
+
+        final String filter = "(username eq \"foo\" and id eq \"bar\") or username eq \"bar\"";
+
+        // one active user
+        final ScimUser scimUser = new ScimUser("bar", "foo", "Some", "Name");
+        scimUser.setOrigin("idp1");
+        final List<ScimUser> expectedUsers = singletonList(scimUser);
+        arrangeScimUsersForFilter(filter, expectedUsers, false, idzId);
+
+        // check different page sizes
+        assertEndpointReturnsCorrectResult(filter, 1, expectedUsers, false);
+        assertEndpointReturnsCorrectResult(filter, 2, expectedUsers, false);
+        assertEndpointReturnsCorrectResult(filter, 3, expectedUsers, false);
+        assertEndpointReturnsCorrectResult(filter, 4, expectedUsers, false);
+        assertEndpointReturnsCorrectResult(filter, 10, expectedUsers, false);
+    }
+
     @Test
     public void testGoodFilter1() {
```

Comment:
Could we get more descriptive test method names instead of numbers on all of these?

Reply:
I migrated the entire test class to JUnit 5. With that, we can use `ParameterizedTest`s to group the different error cases. (see commit 2644f53)